modules/cudev/test/test_nd.cu

```diff
@@ -0,0 +1,247 @@
+#include "test_precomp.hpp"
+
+using namespace cv;
+using namespace cv::cuda;
+using namespace cvtest;
+
+template <typename ElemType>
+class GpuMatNDTest : public ::testing::Test
+{
+public:
+    using MatType = Mat_<ElemType>;
+    using CnType = typename Mat_<ElemType>::channel_type;
+    static constexpr int cn = DataType<ElemType>::channels;
+    using SizeArray = GpuMatND::SizeArray;
+
+    static MatType RandomMat(const SizeArray& size)
+    {
+        const auto dims = static_cast<int>(size.size());
+
+        MatType ret(dims, size.data());
+
+        for (ElemType& elem : ret)
+            for (int i = 0; i < cn; ++i)
+                elem[i] = static_cast<CnType>(std::rand());
```

Comment:
`cv::randu()` ?

Reply:
I have fixed test codes but builds failed because of a merge conflict.